src/fileformats/file_format_registry.cpp

```diff
@@ -97,5 +98,35 @@ const FileFormat *FileFormatRegistry::findFormatForFilename(const QString& filen
 	});
 }
 
+const FileFormat* FileFormatRegistry::findFormatForData(const QString& path, FileFormat::FileTypes types) const
+{
+	QFile file(path);
+	if (!file.open(QIODevice::ReadOnly))
+		return nullptr;
+	
+	char buffer[256];
```

Comment:
Suggest initializing the buffer to zeros (`char buffer[256] = {}`). Number of bytes to be read from the input stream is unknown as well as behavior of underlying `FileFormat::understands()` implementations.

Reply:
But the number of bytes read is known after the next line, and it is passed to `FileFormat::understands()`:
~~~
auto total_read = int(file.read(...));
....
switch (format->understands(buffer, total_read))
...
~~~